Build error with no_std

`cargo build --no-default-features` results in a build error:

```
error[E0433]: failed to resolve: use of undeclared crate or module `std`
   --> src\lib.rs:254:34
    |
254 |             let key = match std::str::from_utf8(key) {
    |                                  ^^^ not found in `std`
    |
help: consider importing this builtin type
    |
32  | use core::primitive::str;
    |
help: if you import `str`, refer to it directly
    |
254 -             let key = match std::str::from_utf8(key) {
254 +             let key = match str::from_utf8(key) {
```

A workflow should be added to test no-std to catch regressions like this in the future.

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -7,6 +7,9 @@ on:
       - ci/**
   pull_request:
 
+env:
+  RUSTFLAGS: -D warnings
+
 jobs:
   build:
     timeout-minutes: 10
@@ -56,13 +59,19 @@ jobs:
         uses: actions-rs/cargo@v1
         with:
           command: clippy
-          args: --target ${{ matrix.target }} -- -D warnings 
+          args: --target ${{ matrix.target }}
           
       - name: build
         uses: actions-rs/cargo@v1
         with:
           command: build
           args: --target ${{ matrix.target }}
+
+      - name: check (no_std)
+        uses: actions-rs/cargo@v1
+        with:
+          command: check
+          args: --target ${{ matrix.target }} --no-default-features
           
       - name: test
         uses: actions-rs/cargo@v1

--- a/src/lib.rs
+++ b/src/lib.rs
@@ -251,7 +251,7 @@ impl<'data> Iterator for KeyValueDataIterator<'data> {
             let key = &key_and_value[..key_end_index];
             let value = &key_and_value[key_end_index + 1..];
 
-            let key = match std::str::from_utf8(key) {
+            let key = match core::str::from_utf8(key) {
                 Ok(key) => key,
                 Err(_) => continue,
             };

